### PR TITLE
Moved Cairo code snippets to Scarb projects in `docs/listings`

### PR DESCRIPTION
**Pull request by integraledelebesgue**
**Stack**:
- #2471
- #2470 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: docs/src/testing/using-cheatcodes.md [hunk 1]
```diff
@@ -19,101 +19,26 @@ using [cheatcodes](../appendix/cheatcodes.md).
...
-    dispatcher.increase_balance(100);
-
-    let balance = dispatcher.get_balance();
-    assert(balance == 100, 'balance == 100');
-}
+{{#include ../../listings/snforge_overview/crates/using_cheatcodes/tests/caller_address/failing.cairo}}
 ```
 
-However, when running this test, we will get a failure with a message
+This test passes, which means that `increase_balance` method panics with expected message.
```
Comment (cptartur) [inline docs/src/testing/using-cheatcodes.md:33]: Why was this changed to should panic?

Comment (integraledelebesgue) [reply docs/src/testing/using-cheatcodes.md:33]: We cannot have panicking tests snce we want to verify our listings using `scarb test`.
Here I made a mistake - `#[should_panic]` attribute is not shown and the test is described as panicking, so I should restore the old text.

Comment (integraledelebesgue) [reply docs/src/testing/using-cheatcodes.md:33]: We cannot have panicking tests since we want to verify Cairo listings using `scarb test`. That's why some tests are marked `#[should_panic]` and some use safe dispatchers.
Also a few tests in the first chapter introducing to testing are presented as failing but in fact they have a hidden `#[should_panic]` not appearing in the text.

File: docs/src/appendix/sncast-library/call.md [hunk 2]
```diff
@@ -6,28 +6,19 @@
...
-use sncast_std::{call, CallResult};
-use starknet::{ContractAddress};
+{{#include ../../../listings/sncast_library/scripts/call/src/lib.cairo}}
+```
 
-fn main() {
-    let contract_address: ContractAddress = 0x1e52f6ebc3e594d2a6dc2a0d7d193cb50144cfdfb7fdd9519135c29b67e427
-        .try_into()
-        .expect('Invalid contract address value');
+Structure use by the command:
```
Comment (cptartur) [inline docs/src/appendix/sncast-library/call.md:17]:
```suggestion
Structure used by the command:
```

File: docs/src/snforge-advanced-features/conditional-compilation.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Conditional Compilation
 
+> 📝 **Note**
+> For an elaborative guide on Scarb conditional compilation, please refer to [Scarb documentation](https://docs.swmansion.com/scarb/docs/reference/conditional-compilation.html)
```
Comment (cptartur) [inline docs/src/snforge-advanced-features/conditional-compilation.md:4]:
```suggestion
> For more detailed guide on Scarb conditional compilation, please refer to [Scarb documentation](https://docs.swmansion.com/scarb/docs/reference/conditional-compilation.html)
```

File: docs/src/snforge-advanced-features/fork-testing.md [hunk 4]
```diff
@@ -94,8 +82,9 @@ Once the fork is configured, the test will run on top of the forked state, meani
...
 
 > ⚠️ **Warning**
-> 
-> The following cheatcodes won't work for forked contracts written in **Cairo 0**:
+>
+> Cheats related to contract execution context won't work for forked contracts written in **Cairo 0**.
+> Those cheatcodes are not going to have any effects:
 >
 > - start_spoof / stop_spoof
 > - spy_events
```
Comment (cptartur) [inline docs/src/snforge-advanced-features/fork-testing.md:90]:
```suggestion
> Cheats related to contract execution context won't work for forked contracts written in **Cairo 0**.
> Those cheatcodes are not going to have any effects:
>
> - `start_spoof` / `stop_spoof`
> - `spy_events`
```

File: docs/src/testing/using-cheatcodes.md (hunk 1 above)
Comment (cptartur) [reply docs/src/testing/using-cheatcodes.md:33]: I've looked at the whole document again and I think this is confusing to have it pass. If that's necessary you can add hidden `should_panic` but we should be showing to the users that this fails so it's obvious what's the point of using a cheat there.

File: docs/src/testing/using-cheatcodes.md [hunk 5]
```diff
@@ -164,42 +71,24 @@ Tests: 1 passed, 0 failed, 0 skipped, 0 ignored, 0 filtered out
 Most cheatcodes come with corresponding `start_` and `stop_` functions that can be used to start and stop the state
 change.
 In case of the `start_cheat_caller_address`, we can cancel the address change
-using [`stop_cheat_caller_address`](../appendix/cheatcodes/caller_address.md#stop_cheat_caller_address)
+using [`stop_cheat_caller_address`](../appendix/cheatcodes/caller_address.md#stop_cheat_caller_address).
+We will demonstrate its behavior using `SafeDispatcher` to show when exactly the fail occurs:
```
Comment (cptartur) [inline docs/src/testing/using-cheatcodes.md:75]: Same here, this should just fail (or at least appear as failed to the user).

File: docs/src/snforge-advanced-features/fork-testing.md (hunk 4 above)
Comment (integraledelebesgue) [reply docs/src/snforge-advanced-features/fork-testing.md:90]: Those information are outdated anyway :)

**Review by cptartur — APPROVED** [review]

**Comment by integraledelebesgue** [comment]
I don't know why is this PR still opened. Closing manually.
